catalog/app/containers/Admin/Status/Status.tsx

```diff
@@ -0,0 +1,487 @@
+import * as R from 'ramda'
+import * as React from 'react'
+import type { ResultOf } from '@graphql-typed-document-node/core'
+import * as M from '@material-ui/core'
+import { fade } from '@material-ui/core/styles'
+import * as MDG from '@material-ui/data-grid'
+
+import * as Chart from 'components/EChartsChart'
+import * as DG from 'components/DataGrid'
+import MetaTitle from 'utils/MetaTitle'
+import useQuery from 'utils/useQuery'
+
+import STATUS_QUERY from './gql/Status.generated'
+
+const STATS_WINDOW = 30
+
+type StatusResult = ResultOf<typeof STATUS_QUERY>['status']
+type Canary = StatusResult['canaries'][number]
+
+interface State {
+  rows: {
+    allRows: string[]
+    idRowsLookup: Record<string, Canary>
+  }
+}
+
+const countsByStateSelector = (state: State) =>
+  state.rows.allRows.reduce(
+    (acc, id) => {
+      const prop = {
+        true: 'passed',
+        false: 'failed',
+        null: 'running',
+      }[`${state.rows.idRowsLookup[id].ok}`]
+      return R.evolve({ [prop]: R.inc }, acc)
+    },
+    { passed: 0, failed: 0, running: 0 },
+  )
+
+const useCountsByStateStyles = M.makeStyles((t) => ({
+  root: {
+    display: 'flex',
+    paddingLeft: t.spacing(2),
+  },
+  item: {
+    alignItems: 'center',
+    display: 'flex',
+
+    '& + &': {
+      marginLeft: t.spacing(2),
+    },
+  },
+  count: {
+    marginLeft: t.spacing(1),
+  },
+}))
+
+function CountsByState() {
+  const classes = useCountsByStateStyles()
+  const apiRef = React.useContext(MDG.GridApiContext)
+  const counts = MDG.useGridSelector(apiRef, countsByStateSelector)
+
+  const renderItem = (count: number, label: string, ok: boolean | null) =>
```

Comment:
Probably, it can be standalone component or standalone function, outside `CountsByState`

Reply:
yeah, it could, but this code is so small and it doesn't make sense on its own